### PR TITLE
Update fuzz target boilerplate to match rust-lightning

### PR DESCRIPTION
**Pull request by TheBlueMatt**
Depends on https://github.com/rust-bitcoin/rust-secp256k1/pull/21.
This should make it easier to have common fuzz infrastructure across rust-bitcoin.

### CODE REVIEW

**Comment by TheBlueMatt** [comment]
Cool, you can see the hfuzz failure in 4c088fc and then the fix in the next commit :).

File: src/util/mod.rs [hunk 1]
```diff
@@ -26,6 +26,7 @@ pub mod hash;
 pub mod iter;
 pub mod misc;
 pub mod uint;
+pub(crate) mod sha2;
```
Comment (apoelstra) [inline src/util/mod.rs]: What does `pub(crate)` mean? Will it increase our minimum compiler version?

Comment (apoelstra) [reply src/util/mod.rs]: Ah, I suppose it exposes this only to the entire crate rather than to downstream users?

Comment (TheBlueMatt) [reply src/util/mod.rs]: Yea, its pub to the crate, the min version for it appears to be 1.18, whats the min version on rust-bitcoin?

Comment (apoelstra) [reply src/util/mod.rs]: After discussion on IRC we landed on 1.14 (currently shipping Debian version), though right now we actually don't compile on 1.14.

However with https://github.com/rust-bitcoin/rust-bech32/pull/14 and a couple minor changes which I will PR presently, we will compile on 1.14.

Comment (TheBlueMatt) [reply src/util/mod.rs]: I'll just make this pub, then. WIthout fuzztarget docs will just say "this reexports crypto::Sha256/Sha512", so thats fine.

**Comment by TheBlueMatt** [comment]
Ok, finally fixed all the pending issues here, I think. Re-tested and you can see the failure on ca52f25 and then fixed on the top commit (date ordering is wrong here...).

File: src/util/sha2.rs [hunk 2]
```diff
@@ -0,0 +1,69 @@
+//! fuzztarget-only Sha2 context with a dummy Sha256 and Sha512 hashers.
+
+#[cfg(feature = "fuzztarget")]
```
Comment (apoelstra) [inline src/util/sha2.rs]: We can drop this (and the `cfg` line further down) since the whole module is feature gated.

Comment (TheBlueMatt) [reply src/util/sha2.rs]: Fixed.

**Pull request merged**